lib/lambda.js

```diff
@@ -50,36 +22,43 @@ module.exports = function elasticApmAwsLambda (agent) {
   }
 
   function wrapContext (trans, payload, context) {
-    const succeed = wrapLambdaCallback(trans, payload, context, (_, res) => context.succeed(res))
-    const done = wrapLambdaCallback(trans, payload, context, (err, res) => context.done(err, res))
-    const fail = wrapLambdaCallback(trans, payload, context, (err) => context.fail(err))
+    shimmer.wrap(context, 'succeed', (succeed) => {
+      return function wrappedSucceed (result) {
+        const bound = succeed.bind(this, result)
+        const done = captureAndMakeCompleter(trans, payload, context, result, bound)
+        done()
+      }
+    })
 
-    return Object.assign({}, context, {
-      succeed: data => succeed(null, data),
-      done: done,
-      fail: fail
+    shimmer.wrap(context, 'fail', (fail) => {
+      return function wrappedFail (err) {
+        const bound = fail.bind(this, err)
+        const done = captureAndMakeCompleter(trans, payload, context, undefined, bound)
+        agent.captureError(err, done)
+      }
+    })
+
+    shimmer.wrap(context, 'done', (done) => {
+      return wrapLambdaCallback(trans, payload, context, done)
     })
   }
 
+  function captureAndMakeCompleter (trans, payload, context, result, callback) {
+    captureContext(trans, payload, context, result)
+    trans.end()
+    return () => {
+      agent.flush(() => callback())
```

Comment:
My initial thought was to log the error here. What's the reason for silencing it?

Reply:
No reason. I can do that too.

Reply:
fixed